smudge-api.el

```diff
@@ -11,84 +11,79 @@
 
 (require 'simple-httpd)
 (require 'request)
+(require 'oauth2)
 
-;; Due to an issue related to compilation and the way oauth2 uses defadvice
-;; (including a FIXME as of 0.1.1), this declaration exists to prevent
-;; compiler warnings.
-(eval-when-compile
-  (defvar url-http-method nil)
-  (defvar url-http-data nil)
-  (defvar url-http-extra-headers nil)
-  (defvar oauth--token-data nil)
-  (defvar url-callback-function nil)
-  (defvar url-callback-arguments nil)
-  (require 'oauth2))
-
-(declare-function oauth2-request-access "oauth2")
-
-(defvar *spotify-user*         nil
-  "Cached user object.")
-(defvar *spotify-oauth2-token* nil
-  "Cached OAuth2 token.")
-(defvar *spotify-oauth2-ts*    nil
-  "Unix timestamp in which the OAuth2 token was retrieved.
-This is used to manually refresh the token when it's about to expire.")
-(defvar *spotify-oauth2-token-file* "~/.emacs.d/.cache/spotify/token"
-	"Location where the OAuth2 token is serialized.")
-
-(defcustom spotify-oauth2-client-id ""
+(defcustom smudge-oauth2-client-id ""
   "The unique identifier for your application.
 More info at https://developer.spotify.com/web-api/tutorial/."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-oauth2-client-secret ""
+(defcustom smudge-oauth2-client-secret ""
   "The OAuth2 key provided by Spotify.
 This is the key that you will need to pass in secure calls to the Spotify
 Accounts and Web API services.  More info at
 https://developer.spotify.com/web-api/tutorial/."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-api-search-limit 50
+(defcustom smudge-api-search-limit 50
   "Number of items returned when searching for something using the Spotify API."
-  :group 'spotify
+  :group 'smudge
   :type 'integer)
 
-(defcustom spotify-api-locale "en_US"
+(defcustom smudge-api-locale "en_US"
   "Optional.  The desired language.
 This consists of an ISO 639 language code and an ISO 3166-1 alpha-2 country
 code, joined by an underscore.  Example: es_MX, meaning Spanish (Mexico).
 Provide this parameter if you want the category metadata returned in a
  particular language."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-api-country "US"
+(defcustom smudge-api-country "US"
   "Optional.  An ISO 3166-1 alpha-2 country code.
 Provide this parameter if you want to narrow the list of returned categories
 to those to a particular country.  If omitted, the returned items will be
 globally relevant."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-oauth2-callback-port "8080"
+(defcustom smudge-oauth2-callback-port "8080"
   "The port for the httpd to listen on for the OAuth2 callback."
-  :group 'spotify
+  :group 'smudge
   :type 'string)
 
-(defcustom spotify-oauth2-callback-endpoint "/spotify-callback"
+(defcustom smudge-oauth2-callback-endpoint "/smudge-callback"
```

Comment:
Is this still necessary? You also introduced a `smudge-api-oauth2-callback` const below.

If this defcustom is not used anymore, we can remove it.

Reply:
Also, one of the screenshots in README is using the other callback URI.

Reply:
@danielfm I think I'll keep it (and use it) in the event someone wants more control over their httpd endpoints. And screenshot is fixed!